### PR TITLE
kube-scheduler's v1 config api is copied for all versions

### PR DESCRIPTION
**Pull request by mvladev**
**How to categorize this PR?**

/area quality
/kind enhancement
/priority normal

**What this PR does / why we need it**:

Scheduler's API is changed between K8S versions. It's better to copy all references.

**Which issue(s) this PR fixes**:

Fixes n/a

**Special notes for your reviewer**:

See #3393

**Release note**:

```other operator
NONE
```

### CODE REVIEW

**Comment by rfranzke** [comment]
/rebase
Can you check the failing CI, please?

File: third_party/kube-scheduler/v18/v1alpha2/types.go [hunk 1]
```diff
@@ -96,7 +96,7 @@ type KubeSchedulerConfiguration struct {
 	// Extenders are the list of scheduler extenders, each holding the values of how to communicate
 	// with the extender. These extenders are shared by all scheduler profiles.
 	// +listType=set
-	Extenders []v1.LegacyExtender `json:"extenders"`
+	Extenders []v1.Extender `json:"extenders"`
```
Comment (timebertt) [inline third_party/kube-scheduler/v18/v1alpha2/types.go:99]: Thanks 👍

**Review by timebertt — DISMISSED** [review]
Thanks for taking care!
/lgtm

File: hack/test-cover.sh [hunk 2]
```diff
@@ -15,8 +15,16 @@
 # limitations under the License.
 set -e
 
+TEST_BIN_DIR="$(dirname "${0}")/../dev/testbin"
```
Comment (timebertt) [inline hack/test-cover.sh:18]: What's the motivation behind the test-related changes apart from bumping the k8s version?

Comment (mvladev) [reply hack/test-cover.sh:18]: It was not up to date with https://github.com/kubernetes-sigs/controller-runtime/blob/master/hack/setup-envtest.sh 

Also, it's better to have it in a local directory than updating system binaries (as it was doing before)

Comment (timuthy) [reply hack/test-cover.sh:18]: If you want to keep it in sync, what do you think about downloading it in the scope of `make revendor`?

Comment (rfranzke) [reply hack/test-cover.sh:18]: @mvladev

**Comment by mvladev** [comment]
I've updated the PR and now it automatically downloads the `setup-envtest.sh` based on the go module version of `controller-runtime`

**Review by timebertt — APPROVED** [review]
Thanks.
/lgtm now

**Review by timuthy — APPROVED** [review]

**Pull request merged**